UPGRADE.md

```diff
@@ -13,3 +13,9 @@
 Also added some files for shared constants (`themeConstants.js`) and useful functions (`themeUtils.js`).
 
 Removed unused custom `createTheme` function in `carto-theme.js`.
+
+# Typography
+
+`responsiveFontSizes` simplified due we want to resize only a few variants through the theme.
+
+Added several custom variants to the typography set. Note that MUI v5 is needed to register them properly.
```

Comment:
Not sure about this comment, we can talk tomorrow "Note that MUI v5 is needed to register them properly"

Reply:
I've added the custom variants, but they can't be used right now because they're not part of the theme.

I have been researching for a long time but I didn't find a way to register them in v4. It seems that custom variants (except colors) are not supported in v4: https://github.com/mui/material-ui/issues/15573

Since this is our next step, I added the new variants and left a note to go ahead and upgrade.